education_evaluation_notebook/models/education_notebook_exam.py

```diff
@@ -0,0 +1,86 @@
+# Copyright (c) 2019 Adrian Revilla <adrianrevilla@avanzosc.es> - Avanzosc S.L.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models, api, exceptions
+
+
+class educationExam(models.Model):
+    _name = 'education.exam'
+    _description = 'Educational exams'
+    
+    name = fields.Char(string='Exam name', required=True)
+    exam_type = fields.Many2one(comodel_name='education.exam.type',
+                                string='Exam type', required=True)
+    n_line = fields.Many2one(comodel_name='education.notebook.line', 
+                             string='Exam asociated competence',
+                             ondelete='cascade')
+    eval_type = fields.Selection(related='n_line.eval_type',
+                                 string='Evaluation season')
+    date = fields.Date(string='Exam date',required=True)
+    eval_percent = fields.Integer(string='Percent (%)',required=True)
+    messages_ok = fields.Char()
+    messages_error = fields.Char()
+    state = fields.Selection([
+            ('notgenerated', 'Expedient not generated'),
+            ('generated', 'Expedient generated'),
+            ],default='notgenerated')
+
+    @api.constrains('eval_percent')
+    def _check_eval_percent(self):
+        for exam in self:
+            exam.clear_messages()
+            if exam.eval_percent < 0 or exam.eval_percent > 100:
+                raise exceptions.ValidationError('Percent incorrect format')
+
+    @api.multi
+    def name_get(self):
+        for exam in self:
+            exam.clear_messages()
+            result = []
+            for record in exam:
+                result.append((record.id, '[Exam] {}'.format(
+                    record.exam_type.name)))
+            return result
+
+    @api.multi
+    def action_generate_expedient(self):
+        for exam in self:
+            exam.clear_messages()
+            exam_id = exam.env['education.exam'].search([('id', '=', exam.id)])
```

Comment:
Al estar haciendo for exam in self, ya tienes el id de examen por lo que el search sobra.
y tampoco necesitarías la variable exam_id, simplemente donde la necesites usas exam.id y ya está

@oihane valida lo que digo si es así.